psycop/common/model_training_v2/loggers/base_logger.py

```diff
@@ -56,3 +59,57 @@ def log_config(self, config: dict[str, Any]) -> None:
         config = flatten_nested_dict(config)
         cfg_str = "\n".join([f"{k}: {v}" for k, v in config.items()])
         self._l.info(cfg_str)
+
+
+class MLFlowLogger(BaselineLogger):
+    def __init__(
+        self,
+        experiment_name: str,
+        tracking_uri: str = "http://exrhel0371.it.rm.dk:5050",
+    ) -> None:
+        mlflow.set_tracking_uri(tracking_uri)
+        mlflow.set_experiment(experiment_name=experiment_name)
+
+        self._log_str = ""
+
+    def _append_log_str(self, prefix: str, message: str):
+        timestamp_str = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self._log_str += f"\n[{timestamp_str}] {prefix}: {message}"
+
+    def _log(self, prefix: str, message: str):
+        """MLFLow supports logging metrics, parameters, datasets and artifacts. Since a log message is neither,
+        the workaround is to create a log file, and then log that as an artifact."""
+
+        self._append_log_str(prefix, message)
+
+        tmp_log_path = Path("tmp_log.txt")
+
+        # Write log temporarily to disk
+        with tmp_log_path.open("w") as f:
+            f.write(self._log_str)
+
+        # Overwrites any previously saved artifacts
+        mlflow.log_artifact(local_path=tmp_log_path.__str__(), artifact_path="log.txt")
+
+        # Delete the log on disk
+        tmp_log_path.unlink()
```

Comment:
A bit hacky here, @KennethEnevoldsen, do we know anyone with experience in mlflow?

Reply:
I sadly don't know anyone